packages/cli/src/lib/option-parsers/client-config.ts

```diff
@@ -1,19 +1,15 @@
-import { validateClientConfig } from "../helpers";
 import { intlMsg } from "../intl";
 import { importTypescriptModule } from "../system";
 import { getTestEnvClientConfig } from "../test-env";
 
-import { Uri } from "@polywrap/core-js";
-import {
-  ClientConfigBuilder,
-  ClientConfig,
-} from "@polywrap/client-config-builder-js";
+import { ClientConfigBuilder } from "@polywrap/client-config-builder-js";
 import path from "path";
+import { IClientConfigBuilder } from "@polywrap/client-config-builder-js";
 
 export async function parseClientConfigOption(
   clientConfig: string | undefined
-): Promise<Partial<ClientConfig<Uri>>> {
-  const builder = new ClientConfigBuilder().addDefaults();
+): Promise<IClientConfigBuilder> {
+  let builder = new ClientConfigBuilder().addDefaults();
```

Comment:
I am probably missing something, but we are adding defaults to the builder, but in every command, we are explicitly adding `noDefaults` to `true`. 

So my question is: Is there any reason for adding defaults?

Reply:
The idea is that the user (in the configure method of the custom config file) can either use the defaults, extend them or override them completely by creating a new instance of IClientConfigBuilder. If we used `noDefaults: false` then the user couldn't override the default resolver configuration.
Also, by returning IClientConfigBuilder instead of something like a union of ClientConfig and CoreClientConfig it leaves the ability for the configuration to be extended further like it's done in the `JobRunner` with its custom config option